airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/python_registry/pipeline.py

```diff
@@ -0,0 +1,165 @@
+        has_setup_py = SETUP_PY_FILE_PATH in files
+        if has_pyproject_toml:
+            return PackageType.POETRY
+        elif has_setup_py:
+            return PackageType.PIP
+        else:
+            return None
+
+    async def _run(self) -> StepResult:
+        package_dir_to_publish = await self.context.get_repo_dir(self.context.package_path)
+        package_type = await self._get_package_type(package_dir_to_publish)
+
+        if not package_type:
+            return self.skip("Connector does not have a pyproject.toml file or setup.py file, skipping.")
+
+        result = await self._ensure_package_name_and_version(package_dir_to_publish, package_type)
+        if result:
+            return result
+
+        self.logger.info(
+            f"Uploading package {self.context.package_metadata.name} version {self.context.package_metadata.version} to {self.context.registry}..."
+        )
+
+        return await self._publish(package_dir_to_publish, package_type)
+
+    async def _ensure_package_name_and_version(self, package_dir_to_publish: Directory, package_type: PackageType) -> Optional[StepResult]:
+        """
+        Try to infer package name and version from the pyproject.toml file. If it is not present, we need to have the package name and version set.
+        Setup.py packages need to set package name and version as parameter.
+
+        Returns None if package name and version are set, otherwise a StepResult with a skip message.
+        """
+        if self.context.package_metadata.name and self.context.package_metadata.version:
+            return None
+
+        if package_type is not PackageType.POETRY:
+            return self.skip("Connector does not have a pyproject.toml file and version and package name is not set otherwise, skipping.")
+
+        inferred_package_metadata = await self._get_package_metadata_from_pyproject_toml(package_dir_to_publish)
+
+        if not inferred_package_metadata:
+            return self.skip(
+                "Connector does not have a pyproject.toml file which specifies package name and version and they are not set otherwise, skipping."
+            )
+
+        if not self.context.package_metadata.name:
+            self.context.package_metadata.name = inferred_package_metadata.name
+        if not self.context.package_metadata.version:
+            self.context.package_metadata.version = inferred_package_metadata.version
+
+        return None
+
+    async def _publish(self, package_dir_to_publish: Directory, package_type: PackageType) -> StepResult:
+        if package_type is PackageType.PIP:
+            return await self._pip_publish(package_dir_to_publish)
+        else:
+            return await self._poetry_publish(package_dir_to_publish)
+
+    async def _poetry_publish(self, package_dir_to_publish: Directory) -> StepResult:
+        pypi_token = self.context.dagger_client.set_secret("pypi_token", f"pypi-{self.context.pypi_token}")
```

Comment:
```suggestion
        python_registy_token = self.context.dagger_client.set_secret("python_registy_token", f"python_registy_token-{self.context.python_registy_token}")
```

Is the `pypi-` prefix required?

Reply:
we could theoretically move it to the secret as well, I'll remove it from here.